components/profile-controller/controllers/profile_controller.go

```diff
@@ -616,15 +624,59 @@ func removeString(slice []string, s string) (result []string) {
 	return
 }
 
+func mergeLabelPropertiesToMap(logger logr.Logger) {
+	file, err := os.Open(NAMESPACE_LABELS_PROPERTIES)
```

Comment:
A controller runs in reconcile loops, this method is called very very often, so it'd be better to load the config file at controller startup and fail the controller if any errors show up.

Reply:
If the configmap is updated, we want to read from the latest data instead of cache for label configuration. So we will need to read from VolumeMount. Is it very resource intense to read a file? If yes, what is your suggestion of reading new ConfigMap data?

Reply:
Note this is a controller, not a service. It's not a big deal to restart an instance to update configs.

Regarding live configmap refresh, I believe there is a config library -- viper which KFP is using too, it can be set-up to watch file/configmap changes, however I think that's not worth the complexity.

Reply:
Thank you for confirming! I think it makes sense and updated the code to reflect one-time file read. I would love to learn about viper later on.